pkg/compose/build_classic.go

```diff
@@ -59,6 +59,12 @@ func (s *composeService) doBuildClassic(ctx context.Context, project *types.Proj
 			errs = multierror.Append(errs, err).ErrorOrNil()
 		}
 		nameDigests[imageName] = digest
+		if errs != nil {
+			return nil
+		}
+		if len(o.Exports) != 0 && o.Exports[0].Attrs["push"] == "true" {
```

Comment:
this really demonstrates we should not use `build.Options ` as input for classic builder, but that's fine for this PR

Reply:
Agree, I'll try to find time to review that